src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,63 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._form = this._popup.querySelector(".modal__form");
+    this._submitButton = this._form.querySelector(".modal__button");
+    this._defaultButtonText = this._submitButton
+      ? this._submitButton.textContent
+      : "Yes";
+  }
+
+  setAction(action) {
+    this._handleSubmit = action;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._form.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleSubmit();
+    });
+  }
+
+  open() {
+    super.open();
+    if (this._submitButton) {
+      this._submitButton.textContent = this._defaultButtonText;
+      this._submitButton.disabled = false;
+    }
+    if (this._resetForm) {
+      this._resetForm();
```

Comment:
You need to **reset** the form only in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.